fontforge/python.c

```diff
@@ -1596,6 +1604,82 @@ Py_RETURN_NONE;
 return( reto );
 }
 
+int NibCheck(SplineSet *ss);
+
+int PyFF_ConvexNibID(char *tok) {
+   int r = ConvexNibID(tok);
+   if ( r==-1 )
+	PyErr_Format(PyExc_TypeError, "Unrecognized convex nib context name" );
+   return r;
+}
+
+static SplineSet *PyFF_ParseSetConvex(PyObject *args, int *tokp) {
+    PyObject *obj;
+    SplineSet *ss;
+    char *tok;
+
+    if ( !PyArg_ParseTuple(args,"O|s", &obj, &tok ) ) {
+	return NULL;
+    }
+    *tokp = PyFF_ConvexNibID(tok);
+    if ( *tokp==-1 )
+	return NULL;
+
+    if ( PyType_IsSubtype(&PyFF_LayerType, Py_TYPE(obj)) ) {
+	ss = SSFromLayer( (PyFF_Layer *) obj);
+    } else if ( PyType_IsSubtype(&PyFF_ContourType, Py_TYPE(obj)) ) {
+	ss = SSFromContour( (PyFF_Contour *) obj, NULL);
+    } else {
+	PyErr_Format(PyExc_TypeError, "Argument must be a layer or a contour" );
+	return( NULL );
+    }
+
+    if ( !NibCheck(ss) ) {
+	SplinePointListFree(ss);
+	return ( NULL );
+    }
+
+    return ss;
+}
+
+static PyObject *PyFF_setConvexNib(PyObject *UNUSED(self), PyObject *args) {
+    SplineSet *ss;
+    int toknum;
+
+    ss = PyFF_ParseSetConvex(args, &toknum);
+    if ( ss==NULL )
+	return NULL;
+
+    if ( toknum < 0 && no_windowing_ui ) {
+	PyErr_Format(PyExc_TypeError, "No user interface" );
+	return NULL;
+    }
+    if ( !StrokeSetConvex(ss, toknum) )
+	return NULL;
+
+    Py_RETURN_NONE;
+}
+
+static PyObject *PyFF_getConvexNib(PyObject *UNUSED(self), PyObject *args) {
+    char *tok;
+    int toknum;
+    SplineSet *ss;
+    PyFF_Layer *l;
+
+    PYGETSTR(args, tok, NULL);
+    toknum = PyFF_ConvexNibID(tok);
+    ENDPYGETSTR();
+    if ( toknum==-1 )
+	return NULL;
+
+    ss = StrokeGetConvex(toknum, false);
+    if ( ss==NULL )
+	Py_RETURN_NONE;
```

Comment:
Seems kind of inconsistent how `return NULL` is used elsewhere then this macro is used here

Reply:
The way this stuff works is that returning a `NULL` means there's an error, while `Py_RETURN_NONE` marks a successful completion for which the python wrapper will return `None`. (Every python function call syntactically returns "something", even if it's just `None`.)

The reason the NULLs "work" here is that they are on conditions that set earlier errors. `PyFF_ConvexNibID()` for example only returns -1 after setting it's own error -- it's actually a wrapper to do just that.